src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java

```diff
@@ -146,6 +146,8 @@ private ShipmentSummaryTO createShipmentSummaryTO(Shipment shipment, Booking boo
   public Mono<BookingTO> createBooking(final BookingTO bookingRequest) {
 
     Booking requestedBooking = bookingMapper.dtoToBooking(bookingRequest);
+    // CarrierBookingRequestReference is not allowed to be set by request
+    requestedBooking.setCarrierBookingRequestReference(null);
```

Comment:
could this be done by adding @Setter(AccessLevel.NONE) to the BookingTO?

Reply:
but then that would prevent a request from PUT from settings it as well,  we accept the CarrierBookingRequestReference  both in path and body and we need to validate that they are the same for a PUT request.